src/openrct2-ui/windows/News.cpp

```diff
@@ -33,316 +33,279 @@ enum WINDOW_NEWS_WIDGET_IDX {
 
+
 static rct_widget window_news_widgets[] = {
     WINDOW_SHIM(WINDOW_TITLE, WW, WH),
     MakeWidget({372, 18}, { 24,  24}, WindowWidgetType::FlatBtn, WindowColour::Primary, SPR_TAB_GEARS_0), // settings
     MakeWidget({  4, 44}, {392, 252}, WindowWidgetType::Scroll,  WindowColour::Primary, SCROLL_VERTICAL), // scroll
     { WIDGETS_END },
 };
 
-static void window_news_mouseup(rct_window *w, rct_widgetindex widgetIndex);
-static void window_news_update(rct_window *w);
-static void window_news_scrollgetsize(rct_window *w, int32_t scrollIndex, int32_t *width, int32_t *height);
-static void window_news_scrollmousedown(rct_window *w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords);
-static void window_news_paint(rct_window *w, rct_drawpixelinfo *dpi);
-static void window_news_scrollpaint(rct_window *w, rct_drawpixelinfo *dpi, int32_t scrollIndex);
-
-static rct_window_event_list window_news_events([](auto& events)
-{
-    events.mouse_up = &window_news_mouseup;
-    events.update = &window_news_update;
-    events.get_scroll_size = &window_news_scrollgetsize;
-    events.scroll_mousedown = &window_news_scrollmousedown;
-    events.paint = &window_news_paint;
-    events.scroll_paint = &window_news_scrollpaint;
-});
-// clang-format on
-
-/**
- *
- *  rct2: 0x0066E464
- */
-rct_window* window_news_open()
+class NewsWindow final : public Window
 {
-    rct_window* window;
-
-    // Check if window is already open
-    window = window_bring_to_front_by_class(WC_RECENT_NEWS);
-    if (window == nullptr)
+private:
+    static int32_t window_news_get_item_height()
     {
-        window = WindowCreateAutoPos(400, 300, &window_news_events, WC_RECENT_NEWS, 0);
-        window->widgets = window_news_widgets;
-        window->enabled_widgets = (1 << WIDX_CLOSE) | (1 << WIDX_SETTINGS);
-        WindowInitScrollWidgets(window);
-        window->news.var_480 = -1;
+        return 4 * font_get_line_height(FontSpriteBase::SMALL) + 2;
     }
 
-    // sub_66E4BA:
-    rct_widget* widget;
-
-    int32_t width = 0;
-    int32_t height = 0;
-    window_get_scroll_size(window, 0, &width, &height);
-    widget = &window_news_widgets[WIDX_SCROLL];
-    window->scrolls[0].v_top = std::max(0, height - (widget->height() - 1));
-    WidgetScrollUpdateThumbs(window, WIDX_SCROLL);
-
-    return window;
-}
-
-static int32_t window_news_get_item_height()
-{
-    return 4 * font_get_line_height(FontSpriteBase::SMALL) + 2;
-}
-
-/**
- *
- *  rct2: 0x0066D4D5
- */
-static void window_news_mouseup(rct_window* w, rct_widgetindex widgetIndex)
-{
-    switch (widgetIndex)
+public:
+    void OnOpen() override
     {
-        case WIDX_CLOSE:
-            window_close(w);
-            break;
-        case WIDX_SETTINGS:
-            context_open_window(WC_NOTIFICATION_OPTIONS);
-            break;
+        widgets = window_news_widgets;
+        enabled_widgets = (1 << WIDX_CLOSE) | (1 << WIDX_SETTINGS);
+        WindowInitScrollWidgets(this);
+        news.var_480 = -1;
```

Comment:
Remove all references to `var_480` and replace it with a new private member field in this class instead. It should be called something like `_pressedNewsItemIndex`.